newParser.cpp

```diff
@@ -142,7 +125,21 @@ std::optional<std::string> commonItems::parser::getNextToken(std::istream& theSt
 
 		if (!matched)
 		{
-			for (auto registration: registeredKeywordRegexes)
+			for (auto registration : registeredKeywordRegexes)
+			{
+				std::smatch match;
+				if (std::regex_match(toReturn, match, std::regex(registration.first)))
```

Comment:
This will generate a regex for every registration for every token (well, the ones not earlier matched) which will be expensive time-wise. Instead, the parser should generate regexes for every  registration when it starts parsing, use those when parsing, then let them fall out of scope and be destroyed.

Reply:
Done so. I wanted to declare generatedRegexes locally within parseStream and then pass the list to getNextToken as argument reference, but this broke Colors::Colors which doesn't care about any matches, so I declared this globally. I've tested across provinces and eu4country and there was no measurable slowdown.